tests/models/soil/test_carbon.py

```diff
@@ -70,7 +72,8 @@ def test_calculate_soil_carbon_updates(dummy_carbon_data, top_soil_layer_index):
     # Check that the updates are correctly calculated
     assert np.allclose(delta_pools[:4], change_in_pools[0])
     assert np.allclose(delta_pools[4:8], change_in_pools[1])
-    assert np.allclose(delta_pools[8:], change_in_pools[2])
+    assert np.allclose(delta_pools[8:12], change_in_pools[2])
+    assert np.allclose(delta_pools[12:], change_in_pools[3])
 
```

Comment:
Do you have to make sure here that the pools are always in the right order or is this just for the test? Would it be worth giving them names and select by name?

Reply:
Yeah the pools should follow the order of the input dictionary, think this should be explicitly tested.

Reply:
I've changed the test setup so that it will fail if the output order does not match the order of the input dictionary